pilot/pkg/model/gateway_test.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"fmt"
+	"testing"
+
+	networking "istio.io/api/networking/v1alpha3"
+)
+
+func makeConfig(name, namespace, host, portName, portProtocol string, portNumber uint32, gw string) Config {
+	c := Config{
+		ConfigMeta: ConfigMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: &networking.Gateway{
+			Selector: map[string]string{"istio": gw},
+			Servers: []*networking.Server{
+				{
+					Hosts: []string{host},
+					Port:  &networking.Port{Name: portName, Number: portNumber, Protocol: portProtocol},
+				},
+			},
+		},
+	}
+	return c
+}
+
+func TestCreateGW(t *testing.T) {
```

Comment:
Looks like the method you're calling is `MergeGateways` ... should the test be named accordingly?

Reply:
Done.